airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/avro_parser.py

```diff
@@ -152,10 +152,18 @@ def file_read_mode(self) -> FileReadMode:
     @staticmethod
     def _to_output_value(avro_format: AvroFormat, record_type: Mapping[str, Any], record_value: Any) -> Any:
         if not isinstance(record_type, Mapping):
-            if record_type == "double" and not avro_format.decimal_as_float:
+            if record_type == "double" and avro_format.double_as_string:
                 return str(record_value)
             return record_value
         if record_type.get("logicalType") == "uuid":
             return uuid.UUID(bytes=record_value)
+        elif record_type.get("logicalType") == "decimal":
+            return str(record_value)
+        elif record_type.get("logicalType") == "date":
```

Comment:
explicitly convert the output to the format we want instead of returning a date/datetime